.github/workflows/ci-aarch64-nightly.yml

```diff
@@ -0,0 +1,142 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# *******************************************************************************
+
+name: "CI AArch64 (Nightly)"
+
+#* To avoid duplicate jobs running when both push and PR is satisfied, we use this:
+#* https://github.com/orgs/community/discussions/26940#discussioncomment-5686753
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  schedule:
+    #* minute (0-59) hour (0-23) day (1-31) month (1-12)  day of the week (0 - 6)
+    #* cron jobs run on the default (main) branch.
+    #* set to run at 5am UCT
+  - cron: "0 5 * * *"
+
+#* Stop stale workflows, though we should never hit this unless it hangs for a whole day.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  build-and-test:
+    strategy:
+      matrix:
+        config: [
+          { name: c6g, label: ah-ubuntu_22_04-c6g_8x-100, threading: OMP, toolset: gcc, build: Release, testset: NIGHTLY },
+          { name: c6g, label: ah-ubuntu_22_04-c6g_8x-100, threading: OMP, toolset: clang, build: Release, testset: NIGHTLY },
+          { name: c7g, label: ah-ubuntu_22_04-c7g_8x-100, threading: OMP, toolset: gcc, build: Release, testset: NIGHTLY },
+          { name: c7g, label: ah-ubuntu_22_04-c7g_8x-100, threading: OMP, toolset: clang, build: Release, testset: NIGHTLY }
+        ]
+
+    name: ${{ matrix.config.name }}, ${{ matrix.config.toolset }}, ${{ matrix.config.threading }}, ${{ matrix.config.build }}
+    runs-on: ${{ matrix.config.label }}
+    steps:
+
+      - name: Checkout oneDNN
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          path: oneDNN
+
+      - name: Get latest CMake and Ninja
+        uses: lukka/get-cmake@8567b9d9b63052b8430ef30042e13c3ba5288f16 # v3.31.3
+        with:
+          cmakeVersion: 3.31.0
+          ninjaVersion: 1.12.0
+
+      - if: ${{ matrix.config.threading == 'OMP' }}
+        name: Install openmp
+        run: |
+          sudo apt install -y libomp-dev
+
+      - if: ${{ matrix.config.toolset == 'gcc' }}
+        name: Install gcc
+        run: |
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
+          sudo apt update -y
+          sudo apt install -y g++-13
+
+      - if: ${{ matrix.config.toolset == 'clang' }}
+        name: Install clang
+        uses: KyleMayes/install-llvm-action@e0a8dc9cb8a22e8a7696e8a91a4e9581bec13181
```

Comment:
Why action? `apt install` should be enough here.

Reply:
This action can install any version of clang on any OS (and can cache if we think it will help). Otherwise we will have to follow manual LLVM instructions for every OS we want to run for to get the specific version we need. For example, clang-17/18/19 can't be taken in ubuntu 22.04 from its default repos.

Reply:
My main concern with this action is that it's hosted under personal account and apparently maintained by a single person.

Reply:
Valid point. Let me see what I can do. 

It will take some time to get this PR ready anyways, because I need to let it run the nightly fully, then I need to get the list of failures and filter them out for the next run, so that the nightly is clean. I can take a look at replacing the action as part of this change as well.

GitHub hosted runners have different versions of clang, but Arm hosted runners have a default OS image, with no clang. I'll play around with a few things. It should not be that difficult, instructions are [here](https://apt.llvm.org). I just need to replace the action with `bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)" 17` I think.

Reply:
Resolved - there is no clang in the nightly anymore, so no need for this action.

For clang in the usual CI, I will deal separately later.